### PR TITLE
feat: support pre-releases 

### PR DESCRIPTION
**Pull request by saintedlama**
A non-non-elegant but working solution

### CODE REVIEW

**Comment by cabauman** [comment]
Nice. I didn't realize you were going to tackle this feature so soon.

**Comment by saintedlama** [comment]
> Nice. I didn't realize you were going to tackle this feature so soon.

I got motivated by having a co-contributor in the project 🥇

File: Versionize/Versioning/SemanticVersionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        var newReleaseLabels = new[] { newPreReleaseLabel, (preReleaseLabel.Equals(newPreReleaseLabel) ? (preReleaseNumber + 1) : 0).ToString() };
+        return new SemanticVersion(version.Major, version.Minor, version.Patch, newReleaseLabels, null);
+    }
+
+    public static SemanticVersion IncrementPatchVersion(this SemanticVersion version)
+    {
+        if (version.IsPrerelease)
+        {
+            // TODO: A bit whacky
```
[the remark this answers is not in the document]
Comment (cabauman) [reply Versionize/Versioning/SemanticVersionExtensions.cs:33]: nitpick: would you mind changing the comment to say what you just said in that reply? For future us 😉 Hopefully we'll eventually be able to narrow down those weird conditions and reproduce.

**Review by cabauman — APPROVED** [review]

**Comment by cabauman** [comment]
Looks good!

**Pull request merged**